### PR TITLE
fix(lua): mark some eval functions that can run in API-fast

### PR DESCRIPTION
**Pull request by lewis6991**
Not exhaustive, but a decent progression.

### CODE REVIEW

File: src/nvim/eval.lua [hunk 1]
```diff
@@ -123,16 +123,16 @@ return {
     feedkeys={args={1, 2}, base=1},
     file_readable={args=1, base=1, func='f_filereadable'},  -- obsolete
     filereadable={args=1, base=1},
```
Comment (folke) [inline src/nvim/eval.lua:125]: Maybe add `filereadable` as well while you're at it :)

File: src/nvim/eval.lua [hunk 2]
```diff
@@ -100,19 +100,19 @@ return {
...
-    escape={args=2, base=1},
+    environ={fast=true},
+    escape={args=2, base=1, fast=true},
     eval={args=1, base=1},
     eventhandler={},
-    executable={args=1, base=1},
+    executable={args=1, base=1, fast=true},
     execute={args={1, 2}, base=1},
     exepath={args=1, base=1},
     exists={args=1, base=1},
```
Comment (folke) [inline src/nvim/eval.lua:118]: `exists` would also be useful, but I'm not sure what works during fast events. Exists checks for files, but also for functions, env vars, commands and autocmds.

**Review by folke — COMMENTED** [review]
Looks good to me. Especially the file (name) related functions are useful to have available during fast events

File: src/nvim/eval.lua (hunk 2 above)
Comment (lewis6991) [reply src/nvim/eval.lua:118]: For now, I've only added functions I'm confident are ok which basically just includes stuff that doesn't depend on any nvim state.

I have no idea if `exists` is ok in fast.

File: src/nvim/eval.lua (hunk 1 above)
Comment (lewis6991) [reply src/nvim/eval.lua]: Done

**Review by bfredl — APPROVED** [review]

File: src/nvim/eval.lua (hunk 2 above)
Comment (folke) [reply src/nvim/eval.lua:118]: Yeah, I agree. Would be hesitant to add that one. Also no idea if it's safe tbh

**Pull request merged**